### PR TITLE
update debuginfo packages (RhBug: 1024701)

### PR DESCRIPTION
**Pull request by MichaelMraka**
enable debuginfo repositories if there are debuginfo packages installed and autoupdate=on

### CODE REVIEW

**Comment by j-mracek** [comment]
Jenkins, test this, please

File: plugins/debuginfo-install.py [hunk 1]
```diff
@@ -34,9 +36,21 @@ class DebuginfoInstall(dnf.Plugin):
...
         if cli is not None:
             cli.register_command(DebuginfoInstallCommand)
 
+    def config(self):
+        cp = self.read_config(self.base.conf, PLUGIN_CONF)
+        autoupdate = cp.getboolean('main','autoupdate')
+
+        if autoupdate:
+            # allow update of already installed debuginfo packages
+            dbginfo = [p.name for p in dnf.sack.rpmdb_sack(self.base).query()
```
Comment (jsilhan) [inline plugins/debuginfo-install.py]: * What's the reason of creating a new sack? `base.sack` is not loaded, yet, right?
* Why the filtering of the name happens afterwards not in the query itself? Not `query().filter(name__glob="*-debuginfo")`
```

File: plugins/dnfpluginscore/lib.py [hunk 2]
```diff
@@ -158,3 +158,9 @@ def package_source_debug_name(package):
     """
     srcname = package_source_name(package)
     return "{}-debuginfo".format(srcname)
+
+def rtrim(s, r):
+    # :api
```
Comment (jsilhan) [inline plugins/dnfpluginscore/lib.py]: I would not add it into API

File: plugins/debuginfo-install.py (hunk 1 above)
Comment (MichaelMraka) [inline plugins/debuginfo-install.py]: >    What's the reason of creating a new sack? base.sack is not loaded, yet, right?

Two reasons - base.sack isn't available at the moment, it's loaded after config and iterating over base.sack.query().installed() is slower than over rpmdb_sack(self.base).query() although they both contain the same set of packages.

>    Why the filtering of the name happens afterwards not in the query itself? Not query().filter(name__glob="*-debuginfo")

I for sure tried this but for a purpose ended up with filtering list; most likely it was slower.
Checking now I can't see a reason for this (maybe something has been fixed in the meantime? :) so changing it to query filter.

**Comment by jsilhan** [comment]
Jenkins retest this please

**Comment by mluscon** [comment]
Most likely not expected behaviour - 
<code>
dnf debuginfo-install xxx
enabling fedora-debuginfo repository
enabling rpmfusion-free-updates-debuginfo repository
enabling rpmfusion-nonfree-updates-debuginfo repository
enabling rpmfusion-free-debuginfo repository
enabling updates-debuginfo repository
enabling rpmfusion-nonfree-debuginfo repository
Traceback (most recent call last):
  File "/home/mluscon/projects/dnf/bin/dnf-2", line 58, in <module>
    main.user_main(sys.argv[1:], exit_code=True)
  File "/home/mluscon/projects/dnf/dnf/cli/main.py", line 174, in user_main
    errcode = main(args)
  File "/home/mluscon/projects/dnf/dnf/cli/main.py", line 60, in main
    return _main(base, args)
  File "/home/mluscon/projects/dnf/dnf/cli/main.py", line 93, in _main
    cli.configure(list(map(ucd, args)))
  File "/home/mluscon/projects/dnf/dnf/cli/cli.py", line 1044, in configure
    self.base.plugins.run_config()
  File "/home/mluscon/projects/dnf/dnf/plugin.py", line 82, in fn
    dnf.util.mapall(operator.methodcaller(method), self.plugins)
  File "/home/mluscon/projects/dnf/dnf/util.py", line 157, in mapall
    return list(map(fn, *seq))
  File "/usr/lib/python2.7/site-packages/dnf-plugins/debuginfo-install.py", line 46, in config
    autoupdate = cp.getboolean('main','autoupdate')
  File "/usr/lib/python2.7/site-packages/iniparse/compat.py", line 146, in getboolean
    v = self.get(section, option)
  File "/usr/lib/python2.7/site-packages/iniparse/compat.py", line 219, in get
    raise NoSectionError(section)
ConfigParser.NoSectionError: No section: 'main'
</code>

**Comment by MichaelMraka** [comment]
Well, it's your misconfiguration because plugin installs valid conf file by default :).

**Pull request merged**